content/doc/developer/plugin-development/split-plugin-from-core.adoc

```diff
@@ -0,0 +1,39 @@
+---
+layout: developer
+title: Split plugin from core
+---
+
+== Introduction
+
+Jenkins has a plugin architecture.
+For historical reasons, some features were developed inside the Jenkins codebase itself (i.e., not as plugins).
+
+To keep that core codebase more manageable, there is an ongoing effort to extract some of the existing features into dedicated plugins.
+
+.Walkthrough video
+video::vGJtbgghYO8[youtube, width=852, height=480, start=480]
+
+== General process to extract classes
+
+* Locate everything that should be moved.
+** Do not forget to look in `core/src/main/resources/` for associated resources.
+*** (`Messages.properties` shared with other classes in the same package can be a headache: you need to create a new `Messages` in a novel package.)
+** And look for relevant functional tests in `test/src/test/`.
+** And icons.
+** And double-check that you have really found everything relevant:
```

Comment:
```suggestion
** Look for relevant functional tests in `test/src/test/`.
** Check for icons.
** Double-check that you have really found everything relevant:
```
small suggestion to update the syntax so that the line does not begin with `And`.

Would the second line also make more sense if it was placed after `Do not forget to look...`?